inference-engine/src/mkldnn_plugin/nodes/mkldnn_tensoriterator_node.cpp

```diff
@@ -196,102 +196,94 @@ class staticValueCheck : public PortChecker {
+         src_stride, dst_stride, m_count, src_stride);
+    m_mem_holder_buffer = new_buffer;
 }
 
-void DynamicBuffer::move_data(mkldnn::stream strm, const mkldnn::engine& eng) {
-    auto axis = map_rule.axis;
-    auto src_mem = from->GetPrimitive();
-    auto src_desc = src_mem.get_desc();
+void DynamicBuffer::move_data() {
+    const auto axis = m_map_rule.m_axis;
+    const auto src_stride = m_from->GetPrimitive().get_desc().dims()[axis] * m_len;
+    const auto dst_stride = m_mem_holder_buffer->get_desc().dims()[axis] * m_len;
 
-    // make chunk view
-    auto chunk_desc = mem_holder_buffer->get_desc();
-    chunk_desc.data.dims[axis] = src_desc.data.dims[axis];
-    chunk_desc.data.padded_dims[axis] = src_desc.data.padded_dims[axis];
+    copy(reinterpret_cast<const uint8_t*>(m_from->GetPtr()), get_ptr(*m_mem_holder_buffer.get()) + m_chunk_offset_in_byte,
+         src_stride, dst_stride, m_count, src_stride);
+}
 
-    const auto new_mem_handler = mem_holder_buffer->get_data_handle();
-    mkldnn::memory chunk_mem = { chunk_desc, eng, new_mem_handler };
-    chunk_mem.set_data_handle(static_cast<uint8_t*>(mem_holder_buffer->get_data_handle()) + chunk_offset_in_byte);
+void DynamicBuffer::transfer(MKLDNNNode* node) {
+    const auto &currDesc = m_to->getDesc();
+    if (!currDesc.getShape().isStatic() || currDesc.getShape().getStaticDims() != m_from->getStaticDims()) {
+        const auto memDesc = node->getBaseMemDescAtOutputPort(m_map_rule.m_from)->cloneWithNewDims(
+                MKLDNNExtensionUtils::convertToVectorDims(m_mem_holder_buffer->get_desc().dims()));
+        m_to->redefineDesc(*memDesc);
+    }
 
-    copy(strm, src_mem, chunk_mem);
+    copy(get_ptr(*m_mem_holder_buffer.get()), reinterpret_cast<uint8_t*>(m_to->GetPtr()), 0, 0, 1, m_to->GetSize());
 }
 
-void DynamicBuffer::transfer(mkldnn::stream strm, const mkldnn::engine& eng, MKLDNNNode* node) {
-    const auto &currDesc = to->getDesc();
-    if (!currDesc.getShape().isStatic() || currDesc.getShape().getStaticDims() != from->getStaticDims()) {
-        const auto memDesc = node->getBaseMemDescAtOutputPort(map_rule.from)->cloneWithNewDims(
-                MKLDNNExtensionUtils::convertToVectorDims(mem_holder_buffer->get_desc().dims()));
-        to->redefineDesc(*memDesc);
-    }
+void DynamicBuffer::copy(const uint8_t* src, uint8_t* dst, const size_t src_stride, const size_t dst_stride, const size_t count, const size_t len) {
+    parallel_for(count, [&](const size_t i) {
+        std::memcpy(&dst[i * dst_stride], &src[i * src_stride], len);
+    });
+}
 
-    auto dst_mem = to->GetPrimitive();
-    copy(strm, *mem_holder_buffer.get(), dst_mem);
+uint8_t* DynamicBuffer::get_ptr(mkldnn::memory& prim) {
+    auto ptr = static_cast<uint8_t*>(prim.get_data_handle());
+    auto md = prim.get_desc().data;
+    mkldnn::impl::memory_desc_wrapper wrapper(md);
+    return ptr + wrapper.offset0() * wrapper.data_type_size();
```

Comment:
I'm not sure about this idea because then we should create `MKLDNNMemoryPtr` as new buffer for each iteration. But for `MKLDNNMemoryPtr` we should have [DnnlBlockedMemoryDesc](https://github.com/openvinotoolkit/openvino/blob/b9f84530daa49ecee7fc98cd0afa3ef92c38487d/inference-engine/src/mkldnn_plugin/mkldnn_memory.h#L90), as I understand. This ctor has a lot of [overheads](https://github.com/openvinotoolkit/openvino/blob/b9f84530daa49ecee7fc98cd0afa3ef92c38487d/inference-engine/src/mkldnn_plugin/memory_desc/dnnl_blocked_memory_desc.cpp#L38) that we don't need (calculations of blocked dims, strides, etc) because `dnnl::memory` is enough for our situation. Moreover these buffers are calculated during inference so we should reduce overheads. 
May be I missed something... Correct me please
But I agree that this `get_ptr` is the same as `GetPtr` of `MKLDNNMemory` and it's not good